components/TextFields/tests/unit/ContainedInputView/MDCBaseTextFieldTests.m

```diff
@@ -184,4 +184,67 @@ - (void)testClearButtonRectForBounds {
   XCTAssertTrue(CGRectEqualToRect(actualClearButtonFrame, expectedClearButtonFrame));
 }
 
+- (void)testFloatingLabelColorAccessors {
+  // Given
+  CGRect textFieldFrame = CGRectMake(0, 0, 130, 40);
+  MDCBaseTextField *textField = [[MDCBaseTextField alloc] initWithFrame:textFieldFrame];
+  UIColor *floatingLabelColor = [UIColor redColor];
+
+  // When
+  [textField setFloatingLabelColor:floatingLabelColor forState:UIControlStateNormal];
```

Comment:
we are only testing this specific combination of these. we should touch all 8 of them.

https://github.com/material-components/material-components-ios/blob/e3784afe259055679479f09b2e92324b61cf7fe0/components/Buttons/tests/unit/MDCButtonTests.m#L383

Since its a discontinuous range we could setup an array of all the bits we care about and loop through them:
controlValues = @[@(Normal),@(Normal |  Editing), ...]

https://github.com/material-components/material-components-ios/blob/e3784afe259055679479f09b2e92324b61cf7fe0/components/Buttons/tests/unit/MDCButtonTests.m#L383

Reply:
Sorry, what are the 8 combinations you're referring to?

`@(Normal | Editing)` isn't a valid state for this textfield. The MDCContainedInputView states are actually mutually exclusive. A contained input view is either normal, disabled, or editing. I actually just made it in an enum in a recent commit for this reason.

Honestly, I still think that using UIControlState for these accessors feels forced. I still think it would be less of a headache down the road to have clients deal with MDContainedInputViewState (even if we renamed it to MDCTextControlState, since MDCContainedInputView is private), the same way we have them deal with MDCTabBarItemState, MDCRippleState, MDCBottomDrawerState, MDCCardCellState, and MDCSheetState.